src/pytorch_lightning/loggers/wandb.py

```diff
@@ -304,7 +304,7 @@ def __init__(
             resume="allow",
             anonymous=("allow" if anonymous else None),
         )
-        self._wandb_init.update(**kwargs)
+        self._wandb_init.update(**{k: v for k, v in kwargs.items() if v is not None})
```

Comment:
Make sure `save_dir` is not overridden by `None` `dir`, see https://github.com/Lightning-AI/lightning/issues/14225